olive/model/__init__.py

```diff
@@ -905,28 +904,37 @@ class DistributedOnnxModel(ONNXModelBase):
 
     def __init__(
         self,
-        model_filepaths: List[Union[Path, str]] = None,
+        model_path: List[Union[Path, str]],
+        model_name_pattern: str,
+        num_ranks: int,
         inference_settings: Optional[dict] = None,
         use_ort_extensions: bool = False,
         model_attributes: Optional[Dict[str, Any]] = None,
     ):
         super().__init__(
-            model_path=None,
+            model_path=model_path,
             inference_settings=inference_settings,
             use_ort_extensions=use_ort_extensions,
             model_attributes=model_attributes,
         )
-        self.model_filepaths = model_filepaths or []
+        self.add_resources({"model_name_pattern": model_name_pattern, "num_ranks": str(num_ranks)})
```

Comment:
`self.add_resources` is only used to add `file/folder` resources. This is so that we can keep track of these for uploading, etc. 
For `model_name_patttern` and `num_ranks`, they should be added directly as attribute during init: `self.model_name_pattern`, etc.

Reply:
Done!